Package:     fox1.6
Subject:     fix secure printing
Author:      Joachim Wiedorn <ad_debian at joonet.de>
Forwarded:   no
Last-Update: 2012-05-17

[Hardening] dpkg-buildflags uses the compiler flag "-Werror=format-security"
by default. This flag shows two errors messages. With this workaround the 
code is more secure.
---

diff -urN s07/src/FXGLViewer.cpp s08/src/FXGLViewer.cpp
--- s07/src/FXGLViewer.cpp	2012-05-17 09:58:12.485142071 +0200
+++ s08/src/FXGLViewer.cpp	2012-05-17 10:03:41.253435120 +0200
@@ -2420,7 +2420,7 @@
 
       // Try open printer
       if(!pdc.beginPrint(printer)){
-        FXMessageBox::error(this,MBOX_OK,tr("Printer Error"),tr("Unable to print."));
+        FXMessageBox::error(this,MBOX_OK,tr("Printer Error"),tr("Unable to print to %s."),"printer");
         return 1;
         }
 
@@ -2595,7 +2595,7 @@
     dlg.getPrinter(printer);
     FXDCPrint pdc(getApp());
     if(!pdc.beginPrint(printer)){
-      FXMessageBox::error(this,MBOX_OK,tr("Printer Error"),tr("Unable to print."));
+      FXMessageBox::error(this,MBOX_OK,tr("Printer Error"),tr("Unable to print to %s."),"printer");
       return 1;
       }
 
